src/components/Register.js

```diff
@@ -0,0 +1,229 @@
+import { setState } from '../actions/index'
+import { register } from '../actions/auth'
+import { connect } from 'react-redux'
+import Button from './Button'
+import Input from './Input'
+import NavBar from './NavBar'
+import OverlaySpinner from './OverlaySpinner'
+import StyledText from './StyledText'
+import Checkbox from './Checkbox'
+import GoogleAnalytics from 'react-native-google-analytics-bridge'
+import { any, isEmpty, converge, isNil, or } from 'ramda'
+
+import { isValidEmail } from '../utils/is-valid-email'
+import { isValidLogin } from '../utils/is-valid-login'
+
+const mapStateToProps = (state) => ({
+  isFetching: state.isFetching,
+  errorMessage: state.errorMessage,
+  isConnected: state.isConnected,
+  registration: state.registration
+})
+
+const mapDispatchToProps = (dispatch) => ({
+  register() {
+    dispatch(register())
+  },
+  setField(fieldName, text) {
+    dispatch(setState(`registration.${fieldName}`, text))
+  },
+  setError(errorMessage) {
+    dispatch(setState('errorMessage', errorMessage))
+  }
+})
+
+export class Register extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {showPasswordText: true}
+    this.handleRegister = this.handleRegister.bind(this)
+    this.handleOpenPrivacyPolicy = this.handleOpenPrivacyPolicy.bind(this)
+  }
+
+  handleRegister() {
+    //prevent red screen of death thrown by a console.error in javascript-client
+    /* eslint-disable no-console */
+    console.reportErrorsAsExceptions = false
+
+    let error  = this.validateForm()
+    this.props.setError(error)
+
+    if (!error) {
+      this.props.register()
+    }
+  }
+
+  validateForm() {
+    const MIN_PASSWORD_LENGTH = 8
+    if (!isValidLogin(this.props.registration.login)) {
+      return 'User Name can\'t contain spaces, dashes or apostrophes'
+    } else if (!isValidEmail(this.props.registration.email)) {
+      return 'Email must be valid'
+    } else if (this.props.registration.password.length < MIN_PASSWORD_LENGTH) {
+      return `Password must be at least ${MIN_PASSWORD_LENGTH} characters long`
+    } else {
+      return null
+    }
+
+  }
+
+  handleOpenPrivacyPolicy() {
+    GoogleAnalytics.trackEvent('view', 'Privacy Policy')
+
+    const url='https://www.zooniverse.org/privacy'
+    Linking.canOpenURL(url).then(supported => {
+      if (supported) {
+        Linking.openURL(url);
+      } else {
+        Alert.alert(
+          'Error', 'Sorry, but it looks like you are unable to open the privacy policy in your default browser.',
+        )
+      }
+    })
+  }
+
+
+  static renderNavigationBar() {
+    return <NavBar showDrawer={false} showBack={true} title={'Register'} />;
+  }
+
+  render() {
+    const registerDisabled = any(converge(or, [isEmpty, isNil]), [this.props.registration.login, this.props.registration.password, this.props.registration.email])
+    const emailChecked = this.props.registration.global_email_communication
```

Comment:
Might be good to give a default prop to `registration`. I can see this throwing an error when trying to look for the `global_email_communication` tag if there isn't a registration prop present.